File: 프로그래머스/lv2/12909. 올바른 괄호/올바른 괄호.js
function solution(s){
    let answer = true;
    let count = 0;
    s.split("").forEach(item => {
        if (item == '(') count++;
        else count--;
        
        if (count < 0) answer = false;
    })
    return answer && count === 0;
    
}

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.py
def solution(s):
    answer = True
    s = [i for i in s]
    stack = []
    
    for c in s:
        if c == "(":
            stack.append(c)
        else:
            if not stack:
                return False
            else:
                stack.pop()
                
        # elif s[i] == ")":
        #     if stack:
        #         stack.pop(-1)
        #     else:
        #         answer = False
        #         break
            
    if stack:
        return False
    

    
    return True